drivers/include/sdcard_spi.h

```diff
@@ -231,8 +231,9 @@ int sdcard_spi_init(sdcard_spi_t *card, const sdcard_spi_params_t *params);
  *
  * @return                number of successfully read blocks (0 if no block was read).
  */
-int sdcard_spi_read_blocks(sdcard_spi_t *card, int blockaddr, uint8_t *data, int blocksize,
-                           int nblocks, sd_rw_response_t *state);
+int sdcard_spi_read_blocks(sdcard_spi_t *card, uint32_t blockaddr,
+                           void *data, uint16_t blocksize,
+                           uint16_t nblocks, sd_rw_response_t *state);
```

Comment:
If we want to support SD cards > 32 GiB, we'll need `uint32_t` for `nblocks`

Reply:
Are you sure? If I understand the code correctly, it is the number of blocks read in one operation to RAM. Since RAM buffers cannot be super large, I think `uint16_t` should be large enough.

A user is more likely to process a large file in one small bit at a time.

Reply:
Ah blockaddr is 32 bit, missed that